indexd/index/drivers/alchemy.py

```diff
@@ -1029,6 +1029,31 @@ def get_latest_version(self, did, has_version=None):
 
             return record.to_document_dict()
 
+    def bulk_get_latest_versions(self, dids, skip_null=False):
+        """
+        Get latest version given list of did
+        """
+        with self.session as session:
+            # get the baseid from given dids to filter the maxdate subquery
+            subq = session.query(IndexRecord.baseid).filter(IndexRecord.did.in_(dids))
+
+            # get max date for each baseid
+            max_date_subq = session.query(IndexRecord.baseid, func.max(IndexRecord.created_date).label('max_date')) \
+                .filter(IndexRecord.baseid.in_(subq)) \
```

Comment:
`.filter(IndexRecord.did.in_(dids))` won't get the latest max date, it will filter on the single did before group by, hence need to filter on baseid from dids passed in.

Reply:
yeah, right. I'm dumb.